File: cpp/gabaritos/gabarito29.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    std::vector<int> lista = {1, 2, 3, 4, 5};
    std::vector<int> quadrados(lista.size());
    std::transform(lista.begin(), lista.end(), quadrados.begin(), [](int x) { return x * x; });
    std::cout << "Quadrados: ";
    for (int num : quadrados) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    return 0;
}

File: csharp/gabaritos/gabarito29.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main()
    {
        List<int> lista = new List<int> { 1, 2, 3, 4, 5 };
        var quadrados = lista.Select(x => x * x).ToList();
        Console.WriteLine("Quadrados: " + string.Join(", ", quadrados));
    }
}